packages/core/src/math/isqrt.rs

```diff
@@ -52,6 +58,29 @@ impl Unsigned for Uint256 {}
 impl Unsigned for Uint512 {}
 impl Unsigned for usize {}
 
+trait Log2 {
+    fn log_2(self) -> u32;
+}
```

Comment:
What about adding this to the (private) `Unsigned` trait? A zero/one could be there as well

Reply:
Oh, I like that way more, especially with the zero and one.